AFQ/api.py

```diff
@@ -269,6 +269,10 @@ def __init__(self,
         '''
         self.logger = logging.getLogger('AFQ.api')
 
+        # validate input and fail early
```

Comment:
Let's separate this argument checking into another pull request(s). I will start writing one for this function now.

Reply:
#456 for AFQ object and issue #450 to remind us to do it elsewhere.